### PR TITLE
refactor(host-infra-system): split git adapter into focused modules and tests

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
This PR splits the `host-infra-system` git adapter into focused modules and reorganizes its test suite to reduce monolithic file size, improve maintainability, and preserve behavior.

## Why
The git system adapter had grown into a large single file (`git.rs`) handling multiple unrelated concerns. A follow-up review also identified that tests had become a monolith after the first split.

### 1) Runtime git adapter modularization
Replaced:
- `apps/desktop/src-tauri/crates/host-infra-system/src/git.rs`

With:
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/mod.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/branch.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/status.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/worktree.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/remote.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/commit.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/util.rs`

Details:
- `GitCliPort` remains the public entry point.
- `impl GitPort for GitCliPort` now delegates to focused internal implementations.
- Shared command plumbing and repository guards remain centralized in `git/mod.rs`.
- Public API compatibility is preserved (`pub use git::GitCliPort` unchanged in `src/lib.rs`).

### 2) Test suite modularization and strengthening
Replaced monolithic test file:
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests.rs`

With focused test modules:
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests/mod.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests/support.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests/branch_tests.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests/commit_tests.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests/remote_tests.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests/status_tests.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests/worktree_tests.rs`
- `apps/desktop/src-tauri/crates/host-infra-system/src/git/tests/validation_tests.rs`

Additional improvements:
- Moved helper/parser-focused tests next to owning source modules:
  - `git/branch.rs`
  - `git/status.rs`
  - `git/util.rs`
- Added parser edge-case coverage for:
  - escaped quoted diff header tokens
  - unterminated quoted diff header tokens
  - malformed ahead/behind token count
  - non-numeric ahead/behind counts

## Behavior and Compatibility
- No contract changes to `GitPort` or host-domain git DTOs.
- No fallback behavior added; failures continue to return actionable errors.
- Existing git command semantics remain unchanged.

## Validation
Executed in `apps/desktop/src-tauri`:
- `cargo fmt --manifest-path crates/host-infra-system/Cargo.toml`
- `cargo test -p host-infra-system`
- `cargo clippy -p host-infra-system --all-targets -- -D warnings`

All checks pass.

## Commits
- `2fc29b56` refactor(host-infra-system): split git adapter into focused modules
- `d8788f1d` test(host-infra-system): split git adapter tests by concern

### CODE REVIEW

File: apps/desktop/src-tauri/crates/host-infra-system/src/git/commit.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        )?;
+        if already_based {
+            return Ok(GitRebaseBranchResult::UpToDate {
+                output: "Branch already contains target history".to_string(),
+            });
+        }
+
+        let (rebase_ok, rebase_stdout, rebase_stderr) =
+            self.run_git_allow_failure(repo_path, &["rebase", target_branch.as_str()])?;
+        let output = combine_output(rebase_stdout, rebase_stderr);
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/crates/host-infra-system/src/git/commit.rs:85]: Implemented in  using  for Current branch vk/f66e-code-quality-spl is up to date. so option parsing is terminated while preserving rebase semantics. Added regression test  to verify option-like values (for example ) are rejected.

[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/crates/host-infra-system/src/git/commit.rs:85]: Implemented in `commit.rs` using `--end-of-options` for `git rebase` so option parsing is terminated while preserving rebase semantics. Added regression test `rebase_branch_rejects_option_like_target` to verify option-like values (for example `--help`) are rejected.

File: apps/desktop/src-tauri/crates/host-infra-system/src/git/remote.rs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            args.push("-u".to_string());
+        }
+        if force_with_lease {
+            args.push("--force-with-lease".to_string());
+        }
+        args.push(remote.clone());
+        args.push(branch.clone());
+
+        let borrowed = args.iter().map(String::as_str).collect::<Vec<_>>();
+        let (ok, stdout, stderr) = self.run_git_allow_failure(repo_path, borrowed.as_slice())?;
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/crates/host-infra-system/src/git/remote.rs:33]: Addressed in `remote.rs` by inserting `--` before `<remote> <branch>` positional operands (after optional flags), preventing option injection through remote/branch names. Added regression test `push_branch_rejects_option_like_remote_input`.

File: apps/desktop/src-tauri/crates/host-infra-system/src/git/branch.rs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        branch: &str,
+        create: bool,
+    ) -> Result<GitCurrentBranch> {
+        self.ensure_repository(repo_path)?;
+        let branch = normalize_non_empty(branch, "branch")?;
+
+        if create {
+            self.run_git(repo_path, &["switch", "-c", branch.as_str()])?;
+        } else {
+            self.run_git(repo_path, &["switch", branch.as_str()])?;
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/crates/host-infra-system/src/git/branch.rs]: Addressed in `branch.rs` by switching non-create path to `git switch --end-of-options <branch>` and adding regression test `switch_branch_rejects_option_like_input`. The create path remains `-c <branch>` intentionally because Git treats that argument as the `-c` operand and rejects invalid refnames rather than parsing it as options.

File: apps/desktop/src-tauri/crates/host-infra-system/src/git/status.rs [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+    pub(super) fn commits_ahead_behind_impl(
+        &self,
+        repo_path: &Path,
+        target_branch: &str,
+    ) -> Result<GitAheadBehind> {
+        self.ensure_repository(repo_path)?;
+        let target = normalize_non_empty(target_branch, "target branch")?;
+        let range = format!("{target}...HEAD");
+        let (ok, stdout, stderr) = self
+            .run_git_allow_failure(repo_path, &["rev-list", "--count", "--left-right", &range])?;
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/crates/host-infra-system/src/git/status.rs]: Addressed in `status.rs` by adding `--end-of-options` to both `git diff` calls and to `git rev-list` argument handling for computed ranges. Added regression test `get_diff_rejects_option_like_target_and_does_not_write_output_file` to guard against option-style payload injection.

File: apps/desktop/src-tauri/crates/host-infra-system/src/git/worktree.rs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    ) -> Result<()> {
+        self.ensure_repository(repo_path)?;
+        let worktree_path = path_to_string(worktree_path, "worktree path")?;
+        let mut args = vec!["worktree".to_string(), "remove".to_string()];
+        if force {
+            args.push("--force".to_string());
+        }
+        args.push(worktree_path);
+        let borrowed = args.iter().map(String::as_str).collect::<Vec<_>>();
+        self.run_git(repo_path, borrowed.as_slice())?;
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/crates/host-infra-system/src/git/worktree.rs:62]: Addressed in `worktree.rs` by adding `--end-of-options` before path operands in create/remove flows and keeping branch/path positional separation explicit. Added regression test `create_worktree_rejects_option_like_branch_input`.

File: apps/desktop/src-tauri/crates/host-infra-system/src/git/branch.rs [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        self.get_current_branch_impl(repo_path)
+    }
+}
+
+pub(super) fn parse_branch_rows(output: &str) -> Vec<GitBranch> {
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/crates/host-infra-system/src/git/branch.rs]: Applied: `parse_branch_rows` is now private (`fn`) in `branch.rs`, reducing visibility to the minimal scope while keeping test coverage intact.

File: apps/desktop/src-tauri/crates/host-infra-system/src/git/status.rs [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+                diff: String::new(),
+            });
+        }
+    }
+
+    results.sort_by(|a, b| a.file.cmp(&b.file));
+    results
+}
+
+pub(super) fn split_diff_by_file(full_diff: &str) -> Vec<(String, String)> {
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/crates/host-infra-system/src/git/status.rs]: Applied: `split_diff_by_file` is now private (`fn`) in `status.rs` since it is only used internally by `build_file_diffs`, reducing module surface area.

**Comment by Maxsky5** [comment]
Follow-up on bot summary comments ([gemini summary](https://github.com/Maxsky5/openducktor/pull/138#issuecomment-3992146798), [greptile summary](https://github.com/Maxsky5/openducktor/pull/138#issuecomment-3992168894)): these are informational overviews, so no code changes were required from them.\n\nActionable review feedback was addressed in commit  with command-injection hardening across git operations, visibility tightening for internal helpers, and new regression tests covering option-like inputs.

**Comment by Maxsky5** [comment]
Correction to my previous note: actionable review feedback was addressed in commit https://github.com/Maxsky5/openducktor/commit/fe0e57e9 with command-injection hardening across git operations, visibility tightening for internal helpers, and regression tests for option-like inputs.
